Model/Api/GetProduct.php

```diff
@@ -65,60 +91,120 @@ class GetProduct implements GetProductInterface
 
     /**
      * @param ProductRepositoryInterface  $productRepositoryInterface
-     * @param StockRegistryInterface  $stockRegistry
+     * @param StockRegistryProviderInterface  $stockRegistry
+     * @param StockConfigurationInterface  $stockConfiguration
      * @param GetProductDataInterface  $productData
      * @param StoreManagerInterface $storeManager
+     * @param Configurable          $configurable
      * @param HookHelper            $hookHelper
      * @param Bugsnag               $bugsnag
      */
     public function __construct(
         ProductRepositoryInterface  $productRepositoryInterface,
-        StockRegistryInterface  $stockRegistry,
+        StockRegistryProviderInterface  $stockRegistry,
+        StockConfigurationInterface  $stockConfiguration,
         GetProductDataInterface  $productData,
         StoreManagerInterface $storeManager,
+        Configurable          $configurable,
         HookHelper $hookHelper,
         Bugsnag $bugsnag
     ) {
         $this->productRepositoryInterface = $productRepositoryInterface;
         $this->stockRegistry = $stockRegistry;
+        $this->stockConfiguration = $stockConfiguration;
         $this->productData = $productData;
         $this->storeManager = $storeManager;
         $this->hookHelper = $hookHelper;
         $this->bugsnag = $bugsnag;
+        $this->configurable = $configurable;
+    }
+
+    private function getStockStatus($product){
+        $stockStatus = $this->stockRegistry->getStockStatus($product->getId(), $this->websiteId);
```

Comment:
`\Magento\CatalogInventory\Api\StockRegistryInterface::getStockStatus` seems to provide even better output, although it is missing individual stock items for each of the stock sources.